test/Tests.Infrastructure/TestBase.cs

```diff
@@ -692,7 +692,6 @@ protected string NewDataPath([CallerMemberName] string prefix = null, string suf
             var path = RavenTestHelper.NewDataPath(prefix, 0, forceCreateDir);
 
             GlobalPathsToDelete.Add(path);
-            _localPathsToDelete.Add(path);
```

Comment:
Does it mean that single test is not cleaning after execution? It is done only after whole suite? If yes, we do not want that.

Reply:
The main issue is that we might have a concurrent file deletion which result in `AccessDenied`, between deleting this at the end of the test and the database deletion.

1. We do clean after each test, each db creation should delete itself.
2. This is why too global place to put paths for deletion, if we have leftovers we need to modify the tests to clean after themselves.

Reply:
1. This is not only used for databases, also used internally for other things like smuggler files, etc. This is the way tests clean after themselves.
2. Global is 'just in case'
3. What do you mean concurrent file deletion? The tests should first delete the database, after that we are cleaning the paths (which in most cases should be already deleted)

Reply:
1. For smuggler and friends `GetTempFileName` should be used so we will add it to `_localPathsToDelete`.
2. You got a point, the concurrent file deletion is only a symptom to the problem.
The deletion is a multi-step operation, we assume that the files are deleted when the command (or the next one) is committed. which is racy and need to be fixed.

So you want me to keep this line after all?